src/genymotion.js

```diff
@@ -0,0 +1,224 @@
+import path from 'path';
+import plist from 'simple-plist';
+import * as registry from 'appcd-winreg';
+
+import { exe } from 'appcd-subprocess';
+import { expandPath } from 'appcd-path';
+import { isDir, isFile } from 'appcd-fs';
+
+/**
+ * Common Genymotion install locations
+ * @type {Object}
+ */
+export const genymotionLocations = {
+	darwin: [
+		'/Applications/Genymotion.app/',
+		'~/Applications/Genymotion.app/'
+	],
+	linux: [
+		'/opt',
+		'/usr',
+		'~'
+	],
+	win32: [
+		'%ProgramFiles%\\Genymobile\\Genymotion',
+		'%ProgramFiles%\\Genymotion',
+		'%ProgramFiles(x86)%\\Genymobile\\Genymotion',
+		'%ProgramFiles(x86)%\\Genymotion'
+	]
+};
+
+/**
+ * Common Genymotion home directory locations
+ * @type {Object}
+ */
+export const genymotionHomeLocations = {
+	darwin: [
+		'~/.Genymobile/Genymotion',
+		'~/.Genymotion'
+	],
+	linux : [
+		'~/.Genymobile/Genymotion',
+		'~/.Genymotion'
+	],
+	win32: [
+		'%LocalAppData%/Genymobile/Genymotion'
+	]
+};
+
+export const genymotionPlist = '~/Library/Preferences/com.genymobile.Genymotion.plist';
+
+/**
+ * Genymotion information
+ */
+export class Genymotion {
+	/**
+	 * Performs tests to see if this is a Genymotion install directory,
+	 * and then initializes the info.
+	 *
+	 * @param {String} dir - Directory to scan.
+	 * @access public
+	 */
+	constructor(dir) {
+		if (typeof dir !== 'string' || !dir) {
+			throw new TypeError('Expected directory to be a valid string');
+		}
+
+		dir = expandPath(dir);
+		if (!isDir(dir)) {
+			throw new Error('Directory does not exist');
+		}
+
+		// on OS X, it lives in Contents/MacOS
+		if (process.platform === 'darwin') {
+			let p = path.join(dir, 'Contents', 'MacOS');
+			if (isDir(p)) {
+				dir = p;
+			} else {
```

Comment:
The `else` is causing issue. Either remove it or possibly change the logic and resolve the Genymotion executable above, then the parent dir is the location of Genymotion.